File: lib/d3c.rb
#!/usr/bin/env ruby
def d3c(text,pw,ty)
	ty=ty.to_s
	if ty!="0" and ty!="1"
		ty="0"
	end
	p=pw.scan /\w/
	if ty=="0"
		pw=0
		p.each { |a| pw+=a.ord }
	elsif ty=="1"
		pw=""
		p.each { |a| pw+=a.ord.to_s }
		pw=pw.to_i
	end
	text=text.to_s
	l=text.scan /\w/
	text=""
	l.each { |a| text+=(a.ord+pw).to_s+";" }
	return text
end
def dd3c(text,pw,ty)
	ty=ty.to_s
	if ty!="0" and ty!="1"
		ty="0"
	end
	p=pw.scan /\w/
	if ty=="0"
		pw=0
		p.each { |a| pw+=a.ord }
	elsif ty=="1"
		pw=""
		p.each { |a| pw+=a.ord.to_s }
		pw=pw.to_i
	end
	l=text.split(";")
	text=""
	l.each { |a| text+=(a.to_i-pw).chr }
	return text
end

File: lib/d3c.py
#!/usr/bin/env python
def en(text,pw,ty):
	ty=str(ty)
	if ty!="0" and ty!="1":
		ty="0"
	p=0 if ty=="0" else ""
	for a in pw:
		p+=ord(a) if ty=="0" else str(ord(a))
	pw=int(p)
	l=[]
	for a in text:
		l.append(str(ord(a)+pw))
	text=";".join(l)
	return text
def de(text,pw,ty):
	ty=str(ty)
	if ty!="0" and ty!="1":
		ty="0"
	p=0 if ty=="0" else ""
	for a in pw:
		p+=ord(a) if ty=="0" else str(ord(a))
	pw=int(p)
	l=[]
	text=text.split(";")
	if text[-1]=="":
		text.pop(-1)
	for a in text:
		l.append(chr(int(a)-pw))
	text="".join(l)
	return text